lib/src/characters_impl.dart

```diff
@@ -90,157 +113,55 @@ class _Characters extends Iterable<String> implements Characters {
     throw RangeError.index(index, this, "index", null, count);
   }
 
+  @override
   bool contains(Object other) {
     if (other is String) {
       if (other.isEmpty) return false;
       int next = Breaks(other, 0, other.length, stateSoTNoBreak).nextBreak();
       if (next != other.length) return false;
```

Comment:
Do we need to reject non-single-grapheme-clusters?
I think it is a foot-gun that `containsAll`, which sounds like an unordered set operation, is what is needed in many cases where the String operation is `contains`.

Reply:
The protocol of `Iterable.contains` is that it answers `true` iff any element is equal to the argument. Since the elements are all single GCs, accepting a string which is not a single GC would break that.

It might be better functionality, but it means that `x.contains(y)` is not the same as `x.any((v) => v == y)` any more. I'm not sure it's a good choice.

It does conflict with `String.contains`, though. Strings are not iterables, so it's not bound by that contract, but it means that `Characters` is different from `String`, which is definitely a user pitfall.


I'm quite willing to change the name of `containsAll`, though. That's definitely confusing. Just haven't found a better name.